src/Mcamara/LaravelLocalization/LaravelLocalization.php

```diff
@@ -226,6 +226,7 @@ public function getLocalizedURL($locale = null, $url = null, $attributes = [], $
         if (empty($attributes)) {
             $attributes = $this->extractAttributes($url, $locale);
         }
+        $urlQuery = parse_url($url, PHP_URL_QUERY);
```

Comment:
You use the expression `($urlQuery ? '?'.$urlQuery : '')` many times.
Perhaps it would be better to calculate it just once.